packages/wallet/test/SingleAddressWallet/load.test.ts

```diff
@@ -165,6 +165,82 @@ const assertWalletProperties2 = async (wallet: ObservableWallet) => {
   expect(rewardAccounts[0].rewardBalance).toBe(mocks.rewardAccountBalance2);
 };
 
+/**
+ * Generates a random byte array in hexadecimal value.
+ *
+ * @param size The size of the hex string in bytes.
+ */
+const genRandomHexString = (size: number) =>
+  [...Array.from({ length: size })]
+    .map(() => Math.floor(Math.random() * 16).toString(16))
+    .join('')
+    .toLowerCase();
+
+/**
+ * Generates a random BigInt number between two values.
+ *
+ * @param min The minimum value.
+ * @param max The maximum value.
+ */
+const generateRandomBigInt = (min: number, max: number): bigint =>
+  BigInt(Math.floor(Math.random() * (max + 1 - min) + min));
+
+/**
+ * Generates a set of UTXOs matching the given parameters.
+ *
+ * @param utxoCount The number of UTXOs to be generated.
+ * @param assetsPerUtxo The number of Assets per UTXO.
+ */
+const generateUtxos = (utxoCount: number, assetsPerUtxo: number): Cardano.Utxo[] => {
+  const utxos: Cardano.Utxo[] = [];
+
+  for (let utxoIndex = 0; utxoIndex < utxoCount; ++utxoIndex) {
+    const utxo: Cardano.Utxo = [
+      {
+        address,
+        index: 0,
+        txId: Cardano.TransactionId(genRandomHexString(64))
+      },
+      {
+        address,
+        value: {
+          assets: new Map(),
+          coins: generateRandomBigInt(1_000_000, 9_999_999_000_000) // from 1 tADA to 9.999.999 tADA
+        }
+      }
+    ];
+    for (let assetIndex = 0; assetIndex < assetsPerUtxo; ++assetIndex) {
+      utxo[1].value!.assets!.set(Cardano.AssetId(genRandomHexString(72)), generateRandomBigInt(1, 1000));
+    }
+
+    utxos.push(utxo);
+  }
+
+  return utxos;
+};
+
+/**
+ * Gets the asset list and the total amount of lovelace on a utxo set.
+ *
+ * @param utxos The utxos.
+ * @returns The total lovelace and the aggregated asset set.
+ */
+const getAssets = (utxos: Cardano.Utxo[]) => {
```

Comment:
```suggestion
const getAssetsFromUtxos = (utxos: Cardano.Utxo[]) => {
```

Better if this was sourced in the `util` package

Reply:
This is actually not possible due to the current dependency `core` has on `util`, so disregard this suggestion. We can look at hoisting when the opportunity arises.